docs/excel/custom-functions-batching.md

```diff
@@ -0,0 +1,214 @@
+2. A function to make the remote request when the batch is ready.
+3. Server code to respond to the batch request, calculate all of the operation results, and return the values.
+
+In the following sections you will be shown how to construct the code one sample at a time. You'll add each code sample to your functions.ts file. It's recommended you create a brand new custom functions project using yo office. To create a new project see [Get started developing Excel custom functions](../quickstarts/excel-custom-functions-quickstart.md).
+
+## Batch each call
+
+You custom functions work by calling a remote service to perform the operation and calculate the result they need. We need to provide a way for them to store each requested operation into a batch. Later you'll see how to create a `_pushOperation` function to batch the operations. First take a look at the following code example to see how to call `_pushOperation` from your custom function.
+
+In the following code the custom function performs division, but relies on a remote service to do the actual calculation. It calls `_pushOperation` to batch the operation along with other operations to the remote service. It names the operation **div2**. You can use any naming scheme you want for operations as long as the remote service is also using the same scheme (more on the remote service later). Also the arguments are passed which the remote service will need to run the operation.
+
+### Add the div2 custom function to functions.ts
+
+```typescript
+/**
+ * @CustomFunction
+ * Divides two numbers using batching
+ * @param dividend The number being divided
+ * @param divisor The number to divide the dividend with
+ * @returns The sum of the two numbers
+ */
+function div2(dividend: number, divisor: number) {
+  return _pushOperation(
+    "div2",
+    [dividend, divisor]
+  );
+}
+
+CustomFunctions.associate("DIV2", div2);
+```
+
+Now to define the batch array which will store all operations to be passed in one network call to the remote service. The following code shows how to define an interface describing each batch entry in the array. The interface defines an operation, which is a string name of which operation to run. For example, if you had two custom functions named `multiply` and `divide`, you could reuse those as the operation names in your batch entries. `args` will hold the arguments that were passed to your custom function from Excel. And finally we store a promise to either `resolve` or `reject` depending on what information the remote service returns.
+
+```typescript
+interface IBatchEntry {
+  operation: string;
+  args: any[];
+  resolve: (data: any) => void;
+  reject: (error: Error) => void;
+}
+```
+
+Next we create the batch array based on the previous interface. Also we create an `_isBatchedRequestSchedule variable to track if a batch is scheduled or not. This will be important later for timing batch calls to the remote service.
```

Comment:
```suggestion
Next, create the batch array that uses the previous interface. To track if a batch is scheduled or not, create an `_isBatchedRequestSchedule variable.  This will be important later for timing batch calls to the remote service.
```